examples/cis2-wccd/src/lib.rs

```diff
@@ -1,37 +1,51 @@
-//! to one correspondence with the CCD.
+//! to one correspondence with the CCD. This smart contract can be
+//! paused/unpaused and upgraded by an admin address.
 //! Note: The word 'address' refers to either an account address or a
 //! contract address.
 //!
-//! As follows from the CIS2 specification, the contract has a `transfer`
+//! As follows from the CIS-2 specification, the contract has a `transfer`
 //! function for transferring an amount of a specific token type from one
 //! address to another address. An address can enable and disable one or more
-//! addresses as operators. An operator of some token owner address is allowed
-//! to transfer any tokens of the owner.
+//! addresses as operators with the `updateOperator` function. An operator of
+//! some token owner address is allowed to transfer or unwrap any tokens of the
+//! owner.
 //!
-//! Besides the contract functions required CIS2, this contract implements a
-//! function `wrap` for converting CCD into wCCD tokens. It accepts an amount of
-//! CCD and mints this amount of wCCD. The function takes a receiving address as
-//! the parameter and transfers the amount of tokens.
+//! Besides the contract functions required by the CIS-2 standard, this contract
+//! implements a function `wrap` for converting CCD into wCCD tokens. It accepts
+//! an amount of CCD and mints this amount of wCCD tokens. The function requires
+//! a receiving address as an input parameter that receives the minted wCCD
+//! tokens.
 //!
 //! The contract also implements a contract function `unwrap` for converting
 //! wCCD back into CCD. The function takes the amount of tokens to unwrap, the
 //! address owning these wCCD and a receiver for the CCD. If the sender is the
-//! owner or an operator of the owner, the wCCD are burned and the amount of
-//! CCD is sent to the receiver.
-
+//! owner or an operator of the owner, the wCCD tokens are burned and the amount
+//! of CCD is sent to the receiver.
+//!
+//! There is a corresponding tutorial for this smart contract available here:
+//! https://developer.concordium.software/en/mainnet/smart-contracts/tutorials/wCCD/index.html
+//!
+//! The admin address can pause/unpause the protocol, set implementors, transfer
+//! the admin address to a new address, and update the metadata URL.
 #![cfg_attr(not(feature = "std"), no_std)]
 use concordium_cis2::*;
-use concordium_std::*;
+use concordium_std::{collections::BTreeMap, *};
 
 /// The id of the wCCD token in this contract.
 const TOKEN_ID_WCCD: ContractTokenId = TokenIdUnit();
 
 /// The metadata url for the wCCD token.
-const TOKEN_METADATA_URL: &str = "https://some.example/token/wccd";
+/// The URL can be updated with the `setURL` function by the `admin`.
+const INITIAL_TOKEN_METADATA_URL: &str = "https://some.example/token/wccd";
+
+/// Tag for the NewAdmin event. The CIS-2 library already uses the
+/// event tags from `u8::MAX` to `u8::MAX - 4`.
+pub const NEW_ADMIN_EVENT_TAG: u8 = u8::MAX - 5;
```

Comment:
Is there a reason for not using `0`. We chose the CIS2 event tags in that way so that other events don't clash with them. And events typically have automatically derived tags, which start with 0...